lbaas-octavia-https/lbstack.yaml

```diff
@@ -0,0 +1,210 @@
+    default: m1.tiny
+    constraints:
+      - custom_constraint: nova.flavor
+  image:
+    type: string
+    description: Image used for servers
+    default: Ubuntu Xenial 16.04 (2020-06-26)
+    constraints:
+      - custom_constraint: glance.image
+  lb_port:
+    type: number
+    default: 80
+    description: Http port used by the load balancer
+  https_lb_port:
+    type: number
+    default: 443
+    description: Https port used by the load balancer
+  ssh_port:
+    type: number
+    default: 22
+    description: SSH port used by the load balancer
+  public_network:
+    type: string
+    description: Network used by the load balancer
+    default: ext-net
+  number_upstreams:
+    type: string
+    default: 4
+  certificate:
+    type: string
+    description: SSL certificate
+    default: db.crt
+  intermediates:
+    type: string
+    description: SSL intermediates
+    default: db.crt
+  private_key:
+    type: string
+    description: SSL key
+    default: db.key
+
+resources:
+
+  syseleven_net:
+    type: OS::Neutron::Net
+    properties:
+      name: syseleven-net
+
+  syseleven_subnet:
+    type: OS::Neutron::Subnet
+    depends_on: [ syseleven_net ]
+    properties:
+      name: syseleven_subnet
+      dns_nameservers:
+        - 8.8.8.8
+        - 4.4.4.4
+      network: {get_resource: syseleven_net}
+      ip_version: 4
+      cidr: 192.168.2.0/24
+      allocation_pools:
+      - {start: 192.168.2.10, end: 192.168.2.250}
+
+  syseleven_router:
+    type: OS::Neutron::Router
+    depends_on: [ syseleven_subnet ]
+    properties:
+      external_gateway_info: {"network": { get_param: public_network }}
+
+  router_subnet_connect:
+    type: OS::Neutron::RouterInterface
+    depends_on: [ syseleven_router ]
+    properties:
+      router: { get_resource: syseleven_router }
+      subnet: { get_resource: syseleven_subnet }
+
+  sec_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      rules:
+      - remote_ip_prefix: 0.0.0.0/0
+        protocol: tcp
+        port_range_min: { get_param: lb_port }
+        port_range_max: { get_param: lb_port }
+      - remote_ip_prefix: 0.0.0.0/0
+        protocol: tcp
+        port_range_min: { get_param: ssh_port }
+        port_range_max: { get_param: ssh_port }
+      - remote_ip_prefix: 0.0.0.0/0
+        protocol: tcp
+        port_range_min: { get_param: https_lb_port }
+        port_range_max: { get_param: https_lb_port }
+
+  monitor:
+    type: OS::Octavia::HealthMonitor
+    depends_on: [ pool ]
+    properties:
+      delay: 3
+      type: HTTP
+      timeout: 3
+      max_retries: 3
+      pool: { get_resource: pool }
+
+  pool:
+    type: OS::Octavia::Pool
+    depends_on: [ listner ]
+    properties:
+      lb_algorithm: ROUND_ROBIN
+      protocol: HTTP
+      listener: { get_resource: listner }
+
+  ssl_certificate:
+    type: OS::Barbican::Secret
+    properties:
+      name: certificate
+      payload: { get_file : { get_param : certificate } }
```

Comment:
get_file on a get_param is not working with my heat version. Heat will never find the files. In my case I needed to remove the parameters and directly hardcode the file names here.

```
pip3 freeze | grep heat
python-heatclient==2.0.0
```

Reply:
true, this does not work.